src/world/partner/watt.c

```diff
@@ -74,12 +91,125 @@ ApiStatus func_802BD27C_31CDEC(Evt* script, s32 isInitialCall) {
+                        D_802BE274_31DDE4->unk_14 = 6.0f;
+                        D_802BE274_31DDE4->unk_18 = 50.0f;
+                        D_802BE274_31DDE4->unk_00 = 0x78;
+                        npc->flags |= 0x40148;
+                        npc->flags &= ~0x200;
+                    case 1:
+                        sin_cos_rad((D_802BE274_31DDE4->unk_10 * TAU) / 360.0f, &sp10, &sp14);
+                        npc->pos.x = (entity->position.x + (sp10 * D_802BE274_31DDE4->unk_0C));
+                        npc->pos.z = (entity->position.z - (sp14 * D_802BE274_31DDE4->unk_0C));
+                        D_802BE274_31DDE4->unk_10 = clamp_angle(D_802BE274_31DDE4->unk_10 - D_802BE274_31DDE4->unk_14);
+
+                        if (D_802BE274_31DDE4->unk_0C > 20.0f) {
+                            D_802BE274_31DDE4->unk_0C--;
+                        } else if (D_802BE274_31DDE4->unk_0C < 19.0f) {
+                            D_802BE274_31DDE4->unk_0C++;
+                        }
+
+                        temp_f0 = sin_rad((D_802BE274_31DDE4->unk_18 * TAU) / 360.0f);
+                        D_802BE274_31DDE4->unk_18 += 3.0f;
+
+                        new_var = temp_f0 * 3.0f;
+
+                        if (D_802BE274_31DDE4->unk_18 > 150.0f) {
+                            D_802BE274_31DDE4->unk_18 = 150.0f;
+                        }
+                        
+                        npc->pos.y += new_var;
+                        npc->renderYaw = clamp_angle(360.0f - D_802BE274_31DDE4->unk_10);
+                        D_802BE274_31DDE4->unk_14 += D_802BE2F0_31DE60;
+
+                        if (D_802BE274_31DDE4->unk_14 > 40.0f) {
+                            D_802BE274_31DDE4->unk_14 = 40.0f;
+                        }
+
+                        if (--D_802BE274_31DDE4->unk_00 == 0) {
+                            D_802BE274_31DDE4->unk_04++;
+                        }
+                        break;
 
-INCLUDE_ASM(s32, "world/partner/watt", func_802BD2B4_31CE24);
+                    case 2:
+                        npc->flags = D_802BE274_31DDE4->unk_08;
+                        D_802BE274_31DDE4->unk_00 = 0x1E;
+                        D_802BE274_31DDE4->unk_04++;
+                        break;
+                    case 3:
+                        partner_flying_update_player_tracking(npc);
+                        partner_flying_update_motion(npc);
+                        if (--D_802BE274_31DDE4->unk_00 == 0) {
+                            D_802BE274_31DDE4->unk_04 = 0;
+                            D_8010C954 = NULL;
+                        }
+
+                        break;
+                }
+            }
+
+        if (D_802BE310 != 0) {
+            D_802BE310->unk_0C->unk_04 = npc->pos.x;
+            D_802BE310->unk_0C->unk_08 = npc->pos.y + 13.0f;
+            D_802BE310->unk_0C->unk_0C = npc->pos.z;
+        }
+
+        return 0;
```

Comment:
This return 0 shouldn't be necessary since we fall into the next one immediately after